src/main/java/net/neoforged/neoforge/common/extensions/IMobEffectExtension.java

```diff
@@ -37,4 +38,11 @@ default void fillEffectCures(Set<EffectCure> cures, MobEffectInstance effectInst
     default int getSortOrder(MobEffectInstance effectInstance) {
         return self().getColor();
     }
+
+    /**
+     * Allow customized attribute level curve
+     */
+    default double getAttributeModifierValue(double amount, int level, ResourceLocation id) {
+        return amount * (double) (level + 1);
+    }
```

Comment:
can this be made attribute sensitive? Potions can have multiple attributes on them and you may want a different curve on each of them

Reply:
Currently I use ResouceLocation for that purpose. Including attribute could be possible, but that could complicate things by a lot.

Reply:
Is that resource location the attribute ID or the attribute modifier ID? Its unclear from this snippet (javadocs should really clarify).

If the ID is the attribute modifier ID then you could just condition over that I suppose, your modifier should be unique if you are adding multiple to the same attribute. They don't actually need to be unique if adding multiple different attributes, though I assume its trivial for someone wishing to do different levels to ensure they are unique. This would also handle the operation case.

If its the attribute ID, then we have no way to know which one between two effects it is.

----

Beyond just differentiation, the other advantage to the instance if its not trivial to fetch is it just saves lookups/comparisons if you want that data. I could imagine my potion might just wish to only scale add differently but leave multiply as unscaled. That would be a lot cleaner code than having to store RL instances and do string comparison against them.

For the attribute class itself, maybe you want to query the range or the max value to use in your formula.

Reply:
Included Holder<Attribute> and Operation as parameter now.